core/src/main/java/io/wcm/testing/mock/aem/MockExternalizer.java

```diff
@@ -142,12 +144,12 @@ private String buildExternalLink(String domain, String scheme, String path) {
     }
 
     // path
-    url.append(path);
+    url.append(getMappedPath(resourceResolver, path));
```

Comment:
@stefanseifert, it appears there is an issue, where it could lead to URL duplication when Sling mappings are configured. In the current implementation, url already contains the base URL, for example, http://localhost:4503. However, the getMappedPath method also returns a fully qualified URL that includes the base URL as part of the Sling mapping for the specified path (e.g., http://localhost:4503/myapp/first-alias/second-alias).

This combination will therefore concatenate these two URLs, resulting in an incorrect URL like http://localhost:4503http://localhost:4503/myapp/first-alias/second-alias.

Reply:
good point - can you create a github issue here for that, and describe your unit test setup you've used to build externalize links with sling mapping? we need to create a test case first, fixing it should be easy.